commit 1c6e3866d3fdc1ed6ea56b6e692e10854789560c
Author: Filip Hanik <fhanik@apache.org>
Date:   Tue May 29 18:04:51 2007 +0000

    only notify the poller if it is done async, otherwise we don't need to, it will be done at the end of the request
    
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@542603 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/coyote/http11/Http11NioProcessor.java b/java/org/apache/coyote/http11/Http11NioProcessor.java
index 9c04224..dca4cb8 100644
--- a/java/org/apache/coyote/http11/Http11NioProcessor.java
+++ b/java/org/apache/coyote/http11/Http11NioProcessor.java
@@ -1227,11 +1227,17 @@ public class Http11NioProcessor implements ActionHook {
             NioEndpoint.KeyAttachment attach = (NioEndpoint.KeyAttachment)socket.getAttachment(false);
             attach.setCometOps(attach.getCometOps()|interest);
             //notify poller if not on a tomcat thread
+            RequestInfo rp = request.getRequestProcessor();
+            if ( rp.getStage() != org.apache.coyote.Constants.STAGE_SERVICE )
+                socket.getPoller().cometInterest(socket);
         } else if (actionCode == ActionCode.ACTION_COMET_UNREGISTER) {
             int interest = getPollerInterest(param);
             NioEndpoint.KeyAttachment attach = (NioEndpoint.KeyAttachment)socket.getAttachment(false);
             attach.setCometOps(attach.getCometOps()& (~interest));
             //notify poller if not on a tomcat thread
+            RequestInfo rp = request.getRequestProcessor();
+            if ( rp.getStage() != org.apache.coyote.Constants.STAGE_SERVICE )
+                socket.getPoller().cometInterest(socket);
         } else if (actionCode == ActionCode.ACTION_COMET_CONFIGURE) {
         }
 
